pkg/controllers/v1alpha1/dataload/dataload_controller.go

```diff
@@ -148,3 +212,25 @@ func (r *DataLoadReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&datav1alpha1.DataLoad{}).
 		Complete(r)
 }
+
+// GetOrCreateEngine gets the Engine
+func (r *DataLoadReconciler) GetOrCreateEngine(
+	ctx cruntime.ReconcileRequestContext) (engine base.Engine, err error) {
+	found := false
+	id := ddc.GenerateEngineID(ctx.NamespacedName)
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	if engine, found = r.engines[id]; !found {
```

Comment:
I am wondering how to clean up the engine if the runtime is deleted.

Reply:
thx, forget to add remove engine when delete, already done